pkg/controllers/workgenerator/controller.go

```diff
@@ -180,7 +179,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 			Message:            "All of the works are synchronized to the latest",
```

Comment:
to make the fix simple, we can remove all the conditions (not set equals unknown) to reset and then add conditions.